model2vec/distill/tokenizer.py

```diff
@@ -109,7 +119,17 @@ def add_tokens(tokenizer: Tokenizer, tokens_to_add: list[str]) -> Tokenizer:
                 wordpiece_vocab[token] = len(wordpiece_vocab)
 
     elif model == "Unigram":
-        raise ValueError("Adding tokens to a unigram tokenizer is not supported.")
+        pre_tokenizer = data["pre_tokenizer"]
+        if pre_tokenizer['type'] != 'Metaspace':
+            if pre_tokenizer['prepend_scheme'] == 'always':
+                unigram_vocab: list[dict[str, Any]] = data["model"]["vocab"]
+                vocab = set(item[0] for item in unigram_vocab)
+                for token in tokens_to_add:
+                    if token not in vocab:
+                        token = pre_tokenizer["replacement"]+token if token[0] != pre_tokenizer["replacement"] else token
```

Comment:
consider just pretokenizing. That would also take care of other issues, e.g., untokenizable strings.